lib/vagrant-registration/action/register.rb

```diff
@@ -42,7 +37,8 @@ def call(env)
 
           @logger.debug('Registration is skipped due to the configuration') if config.skip
 
-          @app.call(env) if Plugin.vbguest_plugin?
+          # Call next middleware in chain
+          @app.call(env)
```

Comment:
> This cannot be done, as it does not work with libvirt provider:

Sure, you need to call the next middleware in chain. I guess the problem is in this case in _plugin.rb_. I'll change back to adding the registration middleware after folder sync for not virtual box.

> However using the different hook for VirtualBox provider could probably work.

I know it does. That's what the tests are for. It should be very easy to write/eable the same tests against libvirt as well.